openlibrary/core/lending.py

```diff
@@ -140,24 +140,43 @@ def get_loan(identifier, user_key=None):
     borrowed that book.
     """
     d = web.ctx.site.store.get("loan-" + identifier)
-    if d and (user_key is None or d['user'] == user_key):
-        loan = Loan(d)
-        if loan.is_expired():
-            loan.delete()
-            return
+
+    account = None
+    if user_key:
+        from openlibrary.accounts.model import OpenLibraryAccount
+        if user_key.startswith('@'):
+            account = OpenLibraryAccount.get(link=user_key)
+        else:
+            account = OpenLibraryAccount.get(username=user_key.split('/')[-1])
+
+    if d:
```

Comment:
I'm not sure I understand this dependency on an OL-db-based loan record. And it seems like this entire conditional branch is *only* used to delete an expired loan. Other than that, it's variables are not used.